rkt/image/resumablesession.go

```diff
@@ -304,6 +307,18 @@ func (s *resumableSession) httpRequest(method string, u *url.URL) *http.Request
 
 	s.setHTTPHeaders(req)
 
+	// Send credentials only over secure channel
+	// TODO(krnowak): This could be controlled with another
+	// insecure flag.
+	if req.URL.Scheme == "https" {
+		if hostOpts, ok := s.Headerers[req.URL.Host]; ok {
+			req = hostOpts.SignRequest(req)
+			if req == nil {
+				panic("Req is nil!")
```

Comment:
If we don't, we'll get a null pointer dereference or whatever Go calls it, or panic, when the req is actually used with the http client.